src/com/mendix/widget/badge/Badge.xml

```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="utf-8"?>
+<widget id="com.mendix.widget.badge.Badge" needsEntityContext="true"
+    xmlns="http://www.mendix.com/widget/1.0/"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../xsd/widget.xsd">
+    <name>Badge</name>
+    <description>Badges are numerical indicators of how many items are associated with a link.</description>
+    <icon>
+    /9j/4AAQSkZJRgABAQEASABIAAD/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAARABwDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAUDBAYH/8QAKRAAAQMDAQcEAwAAAAAAAAAAAQACAwQFESEGEhY1UXORIjFUsYGS4f/EABUBAQEAAAAAAAAAAAAAAAAAAAAB/8QAFBEBAAAAAAAAAAAAAAAAAAAAAP/aAAwDAQACEQMRAD8A6DLI+WRz5HFz3HJJOpKY2mx1t1jfJStYI2HdLnuwCegSwjBIPunFj2gqbRC+KJkckTnb2H50P4RFvg+6dYP3/iVvkrrZLJSmWWFzHepjXaZ6p5xtV/Fg8lZ6tqZ7hVy1UwzJIcndGg0xjwgkvfN6zuu+1SQhFC2WyPK3d0/QQhB//9k=
+    </icon>
+    <properties>
+        <property key="label" type="translatableString" required="false">
+            <caption>Label</caption>
+            <category>Display</category>
+            <description>The label to display before the badge/label</description>
+        </property>
+        <property key="width" type="integer" defaultValue="250">
+            <caption>Width</caption>
+            <category>Display</category>
+            <description>The width in pixels of the widget. If 0 it will be 100%</description>
+        </property>
+        <property key="badgeType" type="enumeration" required="true" defaultValue="badge">
+            <caption>Type</caption>
+            <category>Display</category>
+            <description>Render either the display of a label or of a badge</description>
+            <enumerationValues>
+                <enumerationValue key="badge">badge</enumerationValue>
+                <enumerationValue key="btn">badge button</enumerationValue>
+                <enumerationValue key="label">label</enumerationValue>
+            </enumerationValues>
+        </property>
+        <property key="badgeClass" type="enumeration" required="true" defaultValue="default">
```

Comment:
By default a property is required, no need to specify `required="true"`
Enumerations can't be not required, just like booleans. They have alway a state.